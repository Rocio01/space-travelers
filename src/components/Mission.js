import React from 'react';
import PropTypes from 'prop-types';

/* eslint-disable camelcase */
const Mission = (props) => {
  const {
    mission_id, mission_name, description, status = false,
  } = props;

  return (
    <tr key={mission_id}>
      <td>{mission_id}</td>
      <td>{mission_name}</td>
      <td>{description}</td>
      <td>{status}</td>
    </tr>
  );
};

Mission.propTypes = {
  mission_id: PropTypes.string.isRequired,
  mission_name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  status: PropTypes.bool.isRequired,
};

export default Mission;
