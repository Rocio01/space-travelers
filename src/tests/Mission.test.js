import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Mission from '../components/Mission';
import store from '../redux/configureStore';

test('matches Mission component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Mission />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
