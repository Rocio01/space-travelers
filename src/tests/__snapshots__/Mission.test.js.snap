// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Mission component snapshot 1`] = `
<tr>
  <td />
  <td />
  <td
    className="align-middle text-center"
  >
    <span
      className="btn-secondary span-mission"
    >
      NOT A MEMBER
    </span>
  </td>
  <td
    className="align-middle text-center"
  >
    <button
      className="btn btn-outline-dark mission-button"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;

exports[`matches calculator snapshot 1`] = `
<tr>
  <td />
  <td />
  <td
    className="align-middle text-center"
  >
    <span
      className="btn-secondary span-mission"
    >
      NOT A MEMBER
    </span>
  </td>
  <td
    className="align-middle text-center"
  >
    <button
      className="btn btn-outline-dark mission-button"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
